src/elements/Label/Label.js

```diff
@@ -0,0 +1,168 @@
+/* eslint-disable valid-jsdoc, complexity */
+import cx from 'classnames'
+import React, { createElement, PropTypes } from 'react'
+import createFragment from 'react-addons-create-fragment'
+
+import META from '../../utils/Meta'
+import * as sui from '../../utils/semanticUtils'
+import { someChildType } from '../../utils/childrenUtils'
+import {
+  iconPropRenderer,
+  imagePropRenderer,
+  useKeyOnly,
+  useValueAndKey,
+  useKeyOrValueAndKey,
+} from '../../utils/propUtils'
+import Icon from '../Icon/Icon'
+import Image from '../Image/Image'
+
+/**
+ * A label displays content classification
+ */
+const Label = (props) => {
+  const {
+    attached, children, color, corner, className, circular, detail, detailLink, floating, horizontal, icon,
+    image, link, onClick, onClickDetail, onClickRemove, pointing, removable, ribbon, size, tag, text,
+    ...rest,
+  } = props
+
+  const handleClick = e => onClick && onClick(e, props)
+  const handleClickRemove = e => onClickRemove && onClickRemove(e, props)
+  const handleClickDetail = e => onClickDetail && onClickDetail(e, props)
+
+  const _component = image || link || onClick ? 'a' : 'div'
+
+  const classes = cx('sd-label ui',
+    size,
+    color,
+    useKeyOnly(floating, 'floating'),
+    useKeyOnly(horizontal, 'horizontal'),
+    useKeyOnly(pointing, 'pointing'),
+    useKeyOnly(tag, 'tag'),
+    useValueAndKey(attached, 'attached'),
+    useValueAndKey(corner, 'corner'),
+    useKeyOrValueAndKey(pointing, 'pointing'),
+    useKeyOrValueAndKey(ribbon, 'ribbon'),
```

Comment:
I think the helper methods are valuable. I guess just adding some documentation might help de-obfuscate the syntax. I can just see myself coming back to this code in 6 month and having to re-learn what these helpers are for.